### PR TITLE
feat(scripts): add ROW_COUNT() checks to the procedure-linting script

### PR DESCRIPTION
**Pull request by philbooth**
Related to #411.

In [bug 1499819](https://bugzilla.mozilla.org/show_bug.cgi?id=1499819), it was found that `ROW_COUNT()` causes db replication to lag. As such, it makes sense to grep for `ROW_COUNT()` in our stored procedure lint script. The script was called `scripts/explain-warn.js` before, but in light of the new functionality added here I've renamed it to the more generic `scripts/procedure-lint.js`.

Two procedures fall foul of the new checks, but I couldn't see an obvious way to work round them. In the case of `consumeRecoveryCode_2`, `ROW_COUNT()` is used as an interim signal for aborting the remainder of the procedure. In the case of `setPrimaryEmail_3`, the `UPDATE` is in a transaction which causes `affectedRows` to be zero upon returning.

But still, better to enable linting and ignore those two procedures for now than to not enable linting at all. They'd still remain in the tree without the linting there, all this does is ensure that we'll be warned in the future about any new occurrences of `ROW_COUNT()`, including in subsequent versions of those two procedures.

@mozilla/fxa-devs r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@philbooth - I'm sure this is user error, whenever I run `npm run procedure-lint`, this is the output:

```bash
fxa-auth-db-mysql git:(pb/411) ✗ npm run procedure-lint                                                               git:(pb/411|✚2

> fxa-auth-db-mysql@1.123.2 procedure-lint /Users/shanetomlinson/development/fxa-auth-db-mysql
> node scripts/procedure-lint

{ Error: ENOENT: no such file or directory, open 'lib/db/schema/patch-004-005.sql'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at extractSelects (/Users/shanetomlinson/development/fxa-auth-db-mysql/scripts/procedure-lint.js:327:18)
    at procedures.reduce (/Users/shanetomlinson/development/fxa-auth-db-mysql/scripts/procedure-lint.js:300:7)
    at Array.reduce (<anonymous>)
    at getExplainSmells (/Users/shanetomlinson/development/fxa-auth-db-mysql/scripts/procedure-lint.js:298:30)
    at populateDatabase.then (/Users/shanetomlinson/development/fxa-auth-db-mysql/scripts/procedure-lint.js:87:16)
    at tryCatcher (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/shanetomlinson/development/fxa-auth-db-mysql/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'lib/db/schema/patch-004-005.sql\u001b[36m' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fxa-auth-db-mysql@1.123.2 procedure-lint: `node scripts/procedure-lint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fxa-auth-db-mysql@1.123.2 procedure-lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/shanetomlinson/.fxacache/_logs/2018-10-29T16_28_44_150Z-debug.log
➜  fxa-auth-db-mysql git:(pb/411) ✗
```

**Comment by philbooth** [comment]
> whenever I run `npm run procedure-lint`, this is the output

Weird, in the middle of some Rust stuff at the moment but will dive into this tomorrow.

File: .lint-ignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 accountEmails_4
+consumeRecoveryCode_2
```
Comment (vladikoff) [inline .lint-ignore]: nit: `.lint-ignore` doesn't identify what it is for. is this `.mysql-lint-ignore`? there might be a better name for this

Comment (philbooth) [reply .lint-ignore]: Fair point, will update.

**Comment by philbooth** [comment]
> worked here

Yeah, I fixed the issue Shane raised in a separate PR since it was unrelated to these changes: #425

**Pull request merged**